### PR TITLE
Implemented JPA and Liquibase

### PR DESCRIPTION
**Pull request by MilianCode**

### CODE REVIEW

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
+spring.datasource.url=jdbc:mysql://localhost:3306/online_book_store
 spring.datasource.username=root
 spring.datasource.password=Matepass1!
 spring.jpa.database-platform=org.hibernate.dialect.MySQLDialect
 spring.datasource.driverClassName=com.mysql.cj.jdbc.Driver
 spring.jpa.show-sql=true
 
 server.servlet.context-path=/api
+
+spring.liquibase.change-log=classpath:db/changelog.yml
```
Comment (olekskov) [inline src/main/resources/application.properties]: You should have here master changelog that includes list of patch-sets in it.
In such way it's easier to maintain them when there are more changes

File: src/main/java/com/miliancode/service/impl/BookServiceImpl.java [hunk 2]
```diff
@@ -38,4 +38,20 @@ public BookDto findById(Long id) {
...
     }
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id).get();
```
Comment (olekskov) [inline src/main/java/com/miliancode/service/impl/BookServiceImpl.java]: do not use raw get() on Optional, you make it useless in such way
use orElseThrow()

**Review by olekskov — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 3]
```diff
@@ -67,6 +72,11 @@
                     </excludes>
                 </configuration>
             </plugin>
+            <plugin>
+                <groupId>org.liquibase</groupId>
+                <artifactId>liquibase-maven-plugin</artifactId>
+                <version>4.27.0</version>
```
Comment (Elena-Bruyako) [inline pom.xml:77]: Redundant, spring-boot-starter-parent provide the version of liquibase

File: src/main/java/com/miliancode/model/Book.java [hunk 4]
```diff
@@ -8,8 +9,12 @@
 import java.math.BigDecimal;
 import lombok.Getter;
 import lombok.Setter;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Entity
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id=?")
+@Where(clause = "is_deleted=false")
```
Comment (Elena-Bruyako) [inline src/main/java/com/miliancode/model/Book.java]: `@Where` is deprecated, use SQLRestriction instead

File: src/main/java/com/miliancode/model/Book.java [hunk 5]
```diff
@@ -19,8 +24,11 @@ public class Book {
     private Long id;
     private String title;
```
Comment (Elena-Bruyako) [inline src/main/java/com/miliancode/model/Book.java:26]: `@Column(nullable = false)`
add do each required fields (look at HW task)

File: src/main/java/com/miliancode/service/impl/BookServiceImpl.java [hunk 6]
```diff
@@ -38,4 +38,21 @@ public BookDto findById(Long id) {
...
+
+    @Override
+    public void deleteById(Long id) {
+        bookRepository.deleteById(id);
+    }
+
+    @Override
+    public BookDto update(Long id, CreateBookRequestDto requestDto) {
+        Book book = bookRepository.findById(id)
+                .orElseThrow(() -> new RuntimeException("Book not found with id: " + id));
```
Comment (Elena-Bruyako) [inline src/main/java/com/miliancode/service/impl/BookServiceImpl.java]:
```suggestion
                .orElseThrow(() -> new EntityNotFoundException("Book not found with id: " + id));

```

File: src/main/java/com/miliancode/service/impl/BookServiceImpl.java [hunk 7]
```diff
@@ -38,4 +38,21 @@ public BookDto findById(Long id) {
...
+        book.setTitle(requestDto.getTitle());
```
Comment (Elena-Bruyako) [inline src/main/java/com/miliancode/service/impl/BookServiceImpl.java]: It looks like Mapper should set up Book

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's improve your solution!

**Review by Elena-Bruyako — APPROVED** [review]

**Pull request merged**